src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -113,7 +113,8 @@ Namespace Microsoft.CodeAnalysis.Editor.Implementation.Diagnostics.UnitTests
                 Dim duplicateProjectAnalyzersReference = New AnalyzerImageReference(duplicateProjectAnalyzers)
                 project = project.WithAnalyzerReferences({duplicateProjectAnalyzersReference})
 
-                ' Verify no duplicate descriptors or diagnsotics.
+                ' Verify duplicate descriptors or diagnsotics.
+                ' We don't do de-duplication of analyzer that belong to different layer (host and project)
```

Comment:
that de-duplication will work. so, having same analyzer reference (basically same file) in multiple places regardless of layer (host, project) works.

the one I removed is same instance of diagnostic analyzer in two different analyzer reference. for this to happen, people has to do what the test did. create custom analyzer references and put same instance of diagnostic analyzer in multiple references.

I don't think that is something we need to care.